enum.rb

```diff
@@ -0,0 +1,186 @@
+module Enumerable
+  def my_each
+    return Enumerator unless block_given?
+
+    ar = self
+    ar.each do |el|
+      yield el
+    end
+  end
+
+  def my_each_with_index
+    return Enumerator unless block_given?
+
+    i = 0
+    ar = self
+    ar.my_each do |element|
+      yield i, element
+      i += 1
+    end
+  end
+
+  def my_select
+    return Enumerator unless block_given?
+
+    ar = self
+    new_arr = []
+    ar.my_each do |i|
+      next unless (yield i) == true
+
+      new_arr.append(i)
+    end
+    new_arr
+  end
```

Comment:
* [ ] should have working `my_select` identical and returns the same thing as ruby's [`select`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-select). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_select.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
```ruby
[1,2,3,4,5].my_select != [1,2,3,4,5].select
```

Reply:
Actually when this method don't  have block it return enumerator and original select return enumerator and some memory so we only can return enumerator so this can be the best case

Reply:
same as each and map